packages/unimodules-permissions-interface/ios/UMPermissionsInterface/UMPermissionsInterface.h

```diff
@@ -1,18 +1,37 @@
 // Copyright 2018-present 650 Industries. All rights reserved.
 
 #import <Foundation/Foundation.h>
+#import <UMCore/UMModuleRegistry.h>
+
+typedef enum UMPermissionStatus {
+  UMPermissionStatusDenied,
+  UMPermissionStatusGranted,
+  UMPermissionStatusUndetermined,
+} UMPermissionStatus;
+
+
+@protocol UMPermissionsRequester <NSObject>
+
++ (NSString *)permissionType;
+
+- (void)requestPermissionsWithResolver:(UMPromiseResolveBlock)resolve rejecter:(UMPromiseRejectBlock)reject;
+
+- (NSDictionary *)getPermissions;
+
+@end
 
 @protocol UMPermissionsInterface
 
-- (NSDictionary *)getPermissionsForResource:(NSString *)resource;
-- (BOOL)hasGrantedPermission:(NSString *)permissionType;
+- (void)registerRequesters:(NSArray<id<UMPermissionsRequester>> *)newRequesters;
+
+- (void)getPermissionUsingRequesterClass:(Class)requesterClass
```

Comment:
same comment as above for **both** methods added here -- should be resolver:resolve and rejecter:reject